http/cors.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func wrapCORSHandler(h http.Handler, core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		corsConf := core.CORSConfig()
```

Comment:
@jefferai  is response to your previous comment: OPTIONS (or preflights) requests are made by the browser when a request is made with a method other than `GET`, `HEAD`, or `POST` and the content type is `text/plain`, `application/x-www-form-urlencoded` or `multipart/form-data`.

The preflight request is sent to the API by the browser to determine if the actual request is acceptable to the API. For example when a browser wants to make a `LIST` request the browser will make a preflight request to see if Vault will accept the `LIST` method. 

The preflight response will include the `Access-Control-Max-Age` header, which sets the length of time that the preflight response should be valid so a preflight won't sent before _every_ request.  

Every browser does it a bit differently: Firefox will allow you to set the max age to 24 hours, but Chrome will only allow you to set it to a max of 10 minutes.